subprojects/gst-plugins-bad/sys/msdk/gstmsdkcontextutil.c

```diff
@@ -309,10 +322,42 @@ gst_msdk_context_from_external_display (GstContext * context, gboolean hardware,
   if (ctx)
     return TRUE;
 
-#endif
+  return FALSE;
+}
+#else
+gboolean
+gst_msdk_context_from_external_device (GstContext * context, gboolean hardware,
```

Comment:
Also, I think the name is to genric. Or you change it to _d3d11_device, or just like Haihao said, we need to find a uniform name for both VA and D3D11